nixos/tests/cagebreak.nix

```diff
@@ -0,0 +1,81 @@
+import ./make-test-python.nix ({ pkgs, lib, ...} :
+
+{
+  name = "cagebreak";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ berbiche ];
+  };
+
+  machine = { config, ... }:
+  {
+    imports = [ ./common/user-account.nix ];
+
+    environment.systemPackages = [ pkgs.cagebreak ];
+    services.xserver = {
+      enable = true;
+      displayManager.autoLogin = {
+        enable = true;
+        user = config.users.users.alice.name;
+      };
+    };
+    services.xserver.windowManager.session = lib.singleton {
+      manage = "desktop";
+      name = "cagebreak";
+      start = ''
+        export XDG_RUNTIME_DIR=/run/user/${toString config.users.users.alice.uid}
+        ${pkgs.cagebreak}/bin/cagebreak &
+        waitPID=$!
+      '';
+    };
+
+    systemd.services.setupCagebreakConfig = {
+      wantedBy = [ "multi-user.target" ];
+      before = [ "multi-user.target" ];
+      environment = {
+        HOME = "/home/alice";
+      };
+      unitConfig = {
+        type = "oneshot";
+        RemainAfterExit = true;
+        user = "alice";
+      };
+      script = ''
+        cd $HOME
+        mkdir -p $HOME/.config/cagebreak
+        CONFFILE=$HOME/.config/cagebreak/config
+        cp ${pkgs.cagebreak.contrib}/share/cagebreak/config $CONFFILE
+        chmod 644 $CONFFILE
+        sed -i 's|xterm|${pkgs.xterm}/bin/xterm -cm -pc|g' $CONFFILE
+      '';
+    };
+
+    # Copied from cage:
+    # this needs a fairly recent kernel, otherwise:
+    #   [backend/drm/util.c:215] Unable to add DRM framebuffer: No such file or directory
+    #   [backend/drm/legacy.c:15] Virtual-1: Failed to set CRTC: No such file or directory
+    #   [backend/drm/util.c:215] Unable to add DRM framebuffer: No such file or directory
+    #   [backend/drm/legacy.c:15] Virtual-1: Failed to set CRTC: No such file or directory
+    #   [backend/drm/drm.c:618] Failed to initialize renderer on connector 'Virtual-1': initial page-flip failed
+    #   [backend/drm/drm.c:701] Failed to initialize renderer for plane
+    boot.kernelPackages = pkgs.linuxPackages_latest;
+
+    virtualisation.memorySize = 1024;
+  };
+
+  enableOCR = true;
+
+  testScript = { nodes, ... }: let
+    user = nodes.machine.config.users.users.alice;
+  in ''
+    start_all()
+    machine.wait_for_unit("multi-user.target")
+    # with subtest("ensure cagebreak starts"):
+    machine.wait_for_file("/run/user/${toString user.uid}/wayland-0.lock")
+    # with subtest("ensure xwayland works with xterm"):
+    # machine.send_key("ctrl-t")
+    # machine.send_key("t")
```

Comment:
maybe it would be a good idea to test something that doesn't need xwayland as well

Reply:
Right, I will add a test case with alacritty.